pkg/interface/src/logic/state/base.ts

```diff
@@ -25,10 +26,12 @@ export const reduceState = <
   data: UpdateType,
   reducers: ((data: UpdateType, state: StateType) => StateType)[]
 ): void => {
-  const oldState = state.getState();
-  const reducer = compose(reducers.map(reducer => reducer.bind(reducer, data)));
-  const newState = reducer(oldState);
-  state.getState().set(state => state = newState);
+  unstable_batchedUpdates(() => {
```

Comment:
This should actually be in store/store.ts so that all the updates from a subscription message should be batched together. Also, can you remove the global setState while you're there